src/SmartComponents/BaselinesTable/BaselinesTable.js

```diff
@@ -248,50 +251,93 @@ export class BaselinesTable extends Component {
         return table;
     }
 
+    renderEmptyState = (permissions) => {
+        const { baselineError, emptyState, loading, revertBaselineFetch } = this.props;
+        const { errorMessage } = this.state;
+
+        if (!baselineError.status) {
+            return <EmptyStateDisplay
+                icon={ AddCircleOIcon }
+                title={ EMPTY_BASELINES_TITLE }
+                text={ EMPTY_BASELINES_MESSAGE }
+                button={ <CreateBaselineButton
+                    emptyState={ emptyState }
+                    permissions={ permissions }
+                    loading={ loading } /> }
+            />;
+        } else if (baselineError.status !== 200 && baselineError.status !== undefined) {
+            return <EmptyStateDisplay
+                icon={ ExclamationCircleIcon }
+                color='#c9190b'
+                title={ 'Baselines cannot be displayed' }
+                text={ errorMessage }
+                error={ 'Error ' + baselineError.status + ': ' + baselineError.detail }
+                button={ <a onClick={ () => revertBaselineFetch('CHECKBOX') }>
```

Comment:
`revertBaselineFetch` will need to be passed from the parent components because the `tableId` is going to be different from each parent component. So, in each parent, add `revertBaselineFetch: (tableId) => dispatch(baselinesTableActions.revertBaselineFetch('CHECKBOX'/'COMPARISON'/'RADIO'))` with the argument dependent on the parent. And then when you call `revertBaselineFetch` here, you just call it with `()` and it will pass the proper `tableId` based on the parent component.

I think this will work properly, but not 100% sure.

Reply:
This needs to be changed because if you render the BaselinesTable from AddSystemModal or CreateBaselineModal, then revertBaselineFetch will be called with 'CHECKBOX' every time.